### ISSUE TITLE
How do you validate a Google Pay token for a sandbox?

### ISSUE THREAD

**Issue by xiaolibuzai-ovo** [awa/go-iap#279]
as the title

**Comment by xiaolibuzai-ovo** [comment]
I got a token from the sandbox environment for server-side validation, but got the following error
```
googleapi: Error 400: Invalid Value, invalid
```

I use the >VerifySubscriptionV2

Is it because of the sandbox？

**Comment by richzw** [comment]
> I got a token from the sandbox environment for server-side validation, but got the following error
> 
> ```
> googleapi: Error 400: Invalid Value, invalid
> ```
> 
> I use the VerifySubscriptionV2 api
> 
> Is it because of the sandbox？

No, it is no realted to sandbox. Both the sandbox and production receipt could be verified through `VerifySubscriptionV2` API.

**Comment by xiaolibuzai-ovo** [comment]
> > I got a token from the sandbox environment for server-side validation, but got the following error
> > ```
> > googleapi: Error 400: Invalid Value, invalid
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I use the VerifySubscriptionV2 api
> > Is it because of the sandbox？
> 
> No, it is no realted to sandbox. Both the sandbox and production receipt could be verified through `VerifySubscriptionV2` API.

Thanks for you reply!

What could possibly be causing that? I make sure that packageName exists and that the service account has api access

**Comment by richzw** [comment]
> > > I got a token from the sandbox environment for server-side validation, but got the following error
> > > ```
> > > googleapi: Error 400: Invalid Value, invalid
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > I use the VerifySubscriptionV2 api
> > > Is it because of the sandbox？
> > 
> > 
> > No, it is no realted to sandbox. Both the sandbox and production receipt could be verified through `VerifySubscriptionV2` API.
> 
> Thanks for you reply!
> 
> What could possibly be causing that? I make sure that packageName exists and that the service account has api access

First of all, please make sure the purchase token is correct. Maybe you could find something useful from this link https://stackoverflow.com/questions/46055214/google-play-developer-api-400-invalid-value-inapppurchases.

**Comment by LofiSu** [comment]
You speak English so well
